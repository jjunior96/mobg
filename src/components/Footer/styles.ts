import { darken } from 'polished';
import styled, { css } from 'styled-components';
import media from 'styled-media-query';

import * as HeadingStyles from 'components/Heading/styles';

export const Container = styled.footer`
  ${HeadingStyles.Container} {
    text-transform: uppercase;
  }
`;

export const Content = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: minmax(auto, 50%) 1fr;
    gap: ${theme.grid.gutter};

    margin-top: ${theme.spacings.medium};

    ${media.greaterThan('medium')`
      grid-template-columns: repeat(4, 1fr)
    `}
  `}
`;

export const Column = styled.div`
  ${({ theme }) => css`
    a,
    span {
      display: block;

      color: ${darken(0.2, theme.colors.gray_200)};

      text-decoration: none;

      margin-bottom: ${theme.spacings.xxsmall};

      font-size: ${theme.font.sizes.small};
    }

    a {
      word-wrap: break-word;
      overflow-wrap: break-word;
    }

    a:hover {
      text-decoration: underline;
    }
  `}
`;

export const Copyright = styled.div`
  ${({ theme }) => css`
    color: ${theme.colors.gray_200};
    font-size: ${theme.font.sizes.xxsmall};
    margin-top: ${theme.spacings.large};
    margin-bottom: ${theme.spacings.medium};
    text-align: center;
  `}
`;
