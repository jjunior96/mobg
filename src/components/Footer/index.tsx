import {
  FaInstagram,
  FaTelegram,
  FaFacebook,
  FaWhatsapp
} from 'react-icons/fa';

import Container from 'components/Container';
import Heading from 'components/Heading';
import Input from 'components/Input';
import Logo from 'components/Logo';

import * as S from './styles';

const Footer = () => (
  <S.Container>
    <Container>
      <S.Content>
        <S.Column>
          <Logo />

          <p>Encontre o lugar perfeito para morar</p>

          <ul>
            <li>
              <FaWhatsapp />
            </li>
            <li>
              <FaInstagram />
            </li>
            <li>
              <FaFacebook />
            </li>
            <li>
              <FaTelegram />
            </li>
          </ul>
        </S.Column>

        <S.Column>
          <h3>Navegação</h3>

          <ul>
            <li>Home</li>
            <li>Quero alugar</li>
            <li>Falar com um agente</li>
            <li>Termos de uso</li>
            <li>Política de privacidade</li>
          </ul>
        </S.Column>

        <S.Column>
          <button>Newletter</button>

          <Heading>Receba novidades no seu e-mail</Heading>

          <Input placeholder="Digite seu e-mail" />
        </S.Column>
      </S.Content>
    </Container>

    <Container>
      <S.CopyrightContainer>
        <p>mobg © Todos os direitos reservados</p>

        <div>
          <a href="#">Anunciar imóvel</a>
          <a href="#">Queremos o seu feedback</a>
        </div>
      </S.CopyrightContainer>
    </Container>
  </S.Container>
);

export default Footer;
