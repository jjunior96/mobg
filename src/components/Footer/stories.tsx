import { Story, Meta } from '@storybook/react/types-6-0';

import Footer from '.';

export default {
  title: 'Footer',
  component: Footer
} as Meta;

export const Default: Story = () => (
  <div style={{ marginTop: '1.6rem' }}>
    <Footer />
  </div>
);
