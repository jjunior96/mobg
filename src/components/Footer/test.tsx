import { screen } from '@testing-library/react';

import { renderWithTheme } from 'utils/tests/helpers';

import Footer from '.';

describe('<Footer />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Footer />);

    expect(screen.getByText(/contact/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
