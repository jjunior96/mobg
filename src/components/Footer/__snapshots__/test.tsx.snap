// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render the heading 1`] = `
.c5 {
  color: #292b52;
  font-size: 1.6rem;
}

.c1 {
  font-family: 'Lexend',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-weight: 600;
  font-size: 3.2rem;
  color: #5c63ef;
}

.c0 .c4 {
  text-transform: uppercase;
}

.c2 {
  display: grid;
  grid-template-columns: minmax(auto,50%) 1fr;
  gap: 3.2rem;
  margin-top: 3.2rem;
}

.c3 a,
.c3 span {
  display: block;
  color: #9b9bbc;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 1rem;
  font-size: 2rem;
}

.c3 a {
  word-wrap: break-word;
  overflow-wrap: break-word;
}

.c3 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c6 {
  color: #d8d8e5;
  font-size: 1.4rem;
  margin-top: 4.0rem;
  margin-bottom: 3.2rem;
  text-align: center;
}

@media (min-width:768px) {
  .c2 {
    grid-template-columns: repeat(4,1fr);
  }
}

<footer
  class="c0"
>
  <div
    class=""
  >
    <h2
      class="c1"
      color="primary"
    >
      mobg
    </h2>
  </div>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <h2
        class="c4 c5"
        color="black"
      >
        Contact Us
      </h2>
      <a
        href="mailto:sac@wongames.com"
      >
        sac@wongames.com
      </a>
    </div>
    <div
      aria-labelledby="social-media"
      class="c3"
    >
      <h2
        class="c4 c5"
        color="black"
      >
        Follow us
      </h2>
      <nav
        id="social-media"
      >
        <a
          href="https://www.instagram.com/won-games"
          rel="noopenner, noreferrer"
          target="_blank"
        >
          Instagram
        </a>
        <a
          href="https://www.twitter.com/won-games"
          rel="noopenner, noreferrer"
          target="_blank"
        >
          Twitter
        </a>
        <a
          href="https://www.youtube.com/won-games"
          rel="noopenner, noreferrer"
          target="_blank"
        >
          Youtube
        </a>
        <a
          href="https://www.facebook.com/won-games"
          rel="noopenner, noreferrer"
          target="_blank"
        >
          Facebook
        </a>
      </nav>
    </div>
    <div
      aria-labelledby="resources"
      class="c3"
    >
      <h2
        class="c4 c5"
        color="black"
      >
        Links
      </h2>
      <nav
        id="resources"
      >
        <a
          href="/"
        >
          Home
        </a>
        <a
          href="/games"
        >
          Store
        </a>
        <a
          href="/search"
        >
          Buscar
        </a>
      </nav>
    </div>
    <div
      aria-label="contact"
      class="c3"
    >
      <h2
        class="c4 c5"
        color="black"
      >
        Location
      </h2>
      <span>
        Lorem ipsum dolor sit.
      </span>
      <span>
        Lorem Ipsum
      </span>
      <span>
        Lorem, ipsum dolor.
      </span>
    </div>
  </div>
  <div
    class="c6"
  >
    Won Games 
    2021
     Â© All rights reserved.
  </div>
</footer>
`;
